libfoundation/src/foundation-number.cpp

```diff
@@ -207,6 +245,56 @@ bool MCNumberParseUnicodeChars(const unichar_t *p_chars, uindex_t p_char_count,
 
 ////////////////////////////////////////////////////////////////////////////////
 
+bool MCNumberFormat(MCNumberRef p_number, MCStringRef& r_string)
+{
+    switch(p_number -> flags & kMCNumberFlagRepMask)
+    {
+        case kMCNumberFlagSignedIntegerRep:
+            return MCStringFormat(r_string, "%d", p_number -> integer);
+        case kMCNumberFlagUnsignedIntegerRep:
+            return MCStringFormat(r_string, "%u", p_number -> unsigned_integer);
+        case kMCNumberFlagDoubleRep:
+            return MCStringFormat(r_string, "%lf", p_number -> real);
+    }
+    
+    MCUnreachable();
```

Comment:
I think that stylistically it would be better for this to be in the `switch` statement's `default` branch, but YMMV etc. :wink: